### PR TITLE
Use typed query for manifold-resource-container

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

Converts `<manifold-resource-container>` to use a typed GraphQL query.

This is cleanup work that will make the upcoming ticket https://github.com/manifoldco/engineering/issues/9877 easier.

## Testing

All tests should pass. This doesn’t change any production code; it just makes the types more resilient.

## Checklist

- [ ] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [ ] **Prop changes**: [docs][docs] were updated
- [ ] **Prop changes**: E2E tests were updated, testing from the highest level possible
- [ ] **Platform testing**: If this change should be [tested against a platform][platform-testing], has it been?

[docs]: https://ui.sandbox.manifold.co
[platform-testing]: https://app.gitbook.com/@manifold/s/engineering/playbooks/testing-your-code-against-a-platform

### CODE REVIEW

File: src/data/resource.tsx [hunk 1]
```diff
@@ -1,18 +1,16 @@
 import { h } from '@stencil/core';
 import { createProviderConsumer } from '@stencil/state-tunnel';
-import { Gateway } from '../types/gateway';
-import { Resource } from '../types/graphql';
+import { GetResourceQuery } from '../types/graphql';
 
 export interface ResourceState {
-  data?: Gateway.Resource;
```
Comment (dangodev) [inline src/data/resource.tsx]: This finally removes `data` from the tunnel 🎉  (might rename it in a followup PR, but here I didn’t want to change anything more)

File: src/components/manifold-resource-container/manifold-resource-container.tsx [hunk 2]
```diff
@@ -1,130 +1,12 @@
...
-                  cost
-                }
-              }
-            }
-          }
-        }
-      }
-    }
-  }
-`;
```
Comment (dangodev) [inline src/components/manifold-resource-container/manifold-resource-container.tsx]: This deletes the Gateway mock in `<manifold-resource-mock>`

File: src/components/manifold-resource-plan/manifold-resource-plan.tsx [hunk 3]
```diff
@@ -36,12 +32,13 @@ export class ManifoldResourcePlan {
...
 
+    // TODO: remove `as` and update types
     return (
       <manifold-plan-details
         scrollLocked={false}
-        plan={plan}
-        product={product}
-        region={region}
+        plan={this.gqlData.plan}
+        product={this.gqlData.plan.product as Product}
```
Comment (dangodev) [inline src/components/manifold-resource-plan/manifold-resource-plan.tsx]: This was basically the only temporary hack I had to make. Ideally we’d use generated types for this, too, but that’ll be a whole PR of work once we go down the plan rabbit hole.

But good news is that as long as the shape of `resource.graphql` matches `plan-list.graphql`, we’re fine. But converting that will take some time.

File: src/components/manifold-resource-container/resource.graphql [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+              unit
+              costTiers {
+                limit
+                cost
+              }
+            }
+          }
+        }
+      }
+      configurableFeatures(first: 20) {
```
Comment (sslotsky) [inline src/components/manifold-resource-container/resource.graphql]: I didn't catch this when we first did the conversion, but... what do we use all the regions and features for? Is it for plan resize, or something else?

Comment (dangodev) [reply src/components/manifold-resource-container/resource.graphql]: We need the resource region for display. And we need the features for plan resize. While we probably don’t need to get all regions for resize (you can’t reprovision in a different region), when we convert the plan selector over the types will have to match if we want to reuse the same component.

Comment (sslotsky) [reply src/components/manifold-resource-container/resource.graphql]: When we load the plan selector, we load all of the plans with their features, right? And that includes the selected plan, so we'd be loading that information twice. Maybe I'm wrong but I _think_ the only relevant information that the plan selector needs about the resource is the selected plan ID and region.

Comment (dangodev) [reply src/components/manifold-resource-container/resource.graphql]: We don’t load the plan selector through this; we load `<manifold-plan-details>`. It’s just that `<manifold-plan-details>` needs the exact same type shape, which includes all regions (right now we’re cheating, and using `Plan`, but when we type queries, the types will need to match)

Comment (sslotsky) [reply src/components/manifold-resource-container/resource.graphql]: Ohh, gotcha. In that case, I think we need to put `manifold-plan-details` on the radar for components that need to account for pagination and currently do not. How do we go about doing that? Just have a chat with Meg about it?

Comment (dangodev) [reply src/components/manifold-resource-container/resource.graphql]: Yeah. She started working on this change and has a design somewhere

Comment (sslotsky) [reply src/components/manifold-resource-container/resource.graphql]: Ok. I was aware that she had started this for the plan selector, but was not aware about the plan details. How do we make sure that we are tracking everything that needs pagination support?

Comment (dangodev) [reply src/components/manifold-resource-container/resource.graphql]: Just stay in communication with Meg; she can point you in the right direction in regard to that. That will probably be MUI’s responsibility to figure out a system to track that.

**Pull request merged**